src/transformers/models/vipllava/configuration_vipllava.py

```diff
@@ -88,15 +87,21 @@ def __init__(
         projector_hidden_act="gelu",
         projector_layernorm_eps=1e-5,
         vision_feature_layers=[-2, -5, -8, -11, 6],
-        vocab_size=32000,
         **kwargs,
     ):
         self.ignore_index = ignore_index
         self.image_token_index = image_token_index
         self.projector_hidden_act = projector_hidden_act
         self.projector_layernorm_eps = projector_layernorm_eps
         self.vision_feature_layers = vision_feature_layers
-        self.vocab_size = vocab_size
+
+        if "vocab_size" in kwargs and not isinstance(text_config, dict):
+            warnings.warn(
+                "The `vocab_size` argument is deprecated and will be removed in v4.42, since it can be inferred from the `text_config`.",
+                FutureWarning,
+            )
+            # set the vocab_size
+            text_config["vocab_size"] = kwargs["vocab_size"]
```

Comment:
Accidentally left? This has the same BC issues as before and conflicts with the code below

```suggestion
```